enhancements/sig-policy/72-content-to-ansilbe-job/README.md

```diff
@@ -0,0 +1,165 @@
+launches a predefined Ansible playbook against an inventory of hosts. During this creation, we can pass the 
+additional information to the Ansible job as `extra_vars`.
+
+The current limitation is that we only pass the non-compliant cluster as `extra_vars`, which isn't enough to 
+describe the policy violation status. The following fields could be added:
+1. The violated policy name
+2. The policy violation message
+3. The replicated policy status for each non-compliant cluster
+
+
+### Goals
+
+Adding the name of the violated policy, the violation message, and the replicated policy status for each 
+non-compliant cluster into the Ansible Job.
+
+### Non-Goals
+
+- None at this time
+
+## Proposal
+If policy automation runs on the hub cluster, it should not get the propagated policy status from the managed cluster,
+which requires Search or Observability dependency. The same design also applies to the hub of the cluster scenario.
+
+### New Content Structure
+
+Current `extra_vars` has a field `target_clusters` contains non-compliant cluster name list:
+```json
+{
+  "target_clusters": [
+    "cluster1", 
+    "cluster2", 
+    "cluster3",
+  ]
+}
+```
+
+The new content structure will pass more information:
+```json
+{
+  "target_clusters": [
+    "cluster1", 
+    "cluster2", 
+    "cluster3",
+  ],
+  "policy_name": "policy1",
+  "violation_message": "NonCompliant; violation -  xxxxxx",
+  "policy_violation_context" : {
+    "cluster1": "raw replicated policy status1",
+    "cluster2": "raw replicated policy status2",
+    },
+}
+```
+
+- `policy_name` is the name of the non-compliance policy that triggers the ansible job.
+  `violation_message` is the latest policy violation message. 
+
+  Suppose the policy automation runs on the hub cluster. In that case, `policy_name` and `violation_message` are from 
+  the roof policy. 
+
+  Suppose the policy automation runs on the managed cluster. In that case, `policy_name` and `violation_message` are from the propagated policy on the managed cluster.
+
+  The field path of `policy_name` is `policy.metadata.name`.
+  The field path of `violation_message` is `policy.policyStatus.details[0].history.message[0]`
+
+- `policy_violation_context` is a map. The keys are non-compliant cluster names, and the value is the raw replicated 
+  policy status for each non-compliant cluster.
+
+  Suppose the policy automation runs on the hub cluster. In that case, the policy automation looks at the `cluster namespaces` 
+  of the hub cluster and gets the replicated policies. Then it filters out the non-compliance replicated policies and puts the raw status into the `policy_violation_context` map.
+
+  Suppose the policy automation runs on the managed cluster. In that case, the policy automation looks at the propagated policy on the managed cluster, then `policy_violation_context` would be a map that only contains the propagated policy status.
+
+  The field path of the raw status is `policy.policyStatus`.
```

Comment:
Is there any reason for users to get the entire raw status? I was thinking we would at least cut out the `status.details[n].templateMeta` field, since that is only used by the controller to maintain prune behavior. We also probably don't need `eventName` in `history`.